src/MovieCard.jsx

```diff
@@ -1,27 +1,72 @@
-// import { useState } from 'react'
 import './MovieCard.css'
 import PropTypes from 'prop-types'
 
-const MovieCard = ({id, title, rating, poster_path, onMovieCardClick}) => {
+const MovieCard = ({id, title, rating, poster_path, onMovieCardClick, onHandleLikeClick, onHandleWatchedClick}) => {
 
   const onCardClick = () => {
     document.getElementById('modal-overlay').style.display = 'block';
     document.getElementById('modal-overlay').scrollIntoView({behavior: 'smooth'});
     onMovieCardClick(id);
   }
 
+  let clickedLikeBefore = false;
```

Comment:
- This is a perfect candidate for `useState` (and so is the watchedBefore)
- When in state, you could simplify the `onClick` like below
- sidenote: `setState` functions can instead have a function as a parameter which gives you the previous value

```
const [isLiked, setIsLiked] = useState(false);
const handleOnClick = (event) => {
  event.stopPropagation();
  setIsLiked(prevIsLiked => {
    onHandleLikeClick(simplifiedMovieObject, !prevIsLiked);
    return !prevIsLiked;
  });
}
```
Then use the `isLiked` value to change the rendering rather than updating the actual dom element

```
<span className={isLiked ? "like-icon-liked" : "like-icon-unliked"} onClick={handleLikeClick}>♥</span>
```